app/connectors/VatRegistrationConnector.scala

```diff
@@ -243,6 +243,11 @@ class VatRegistrationConnector @Inject()(val http: HttpClient,
     val js = Json.parse(s"""{"transactionID": "$transactionId"}""").as[JsObject]
     http.PATCH[JsObject, HttpResponse](s"$vatRegUrl/vatreg/$regId/transaction-id", js)
   }
+
+  def submitHonestyDeclaration(regId: String, honestyDeclaration: Boolean)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
+    val js = Json.parse(s"""{"honestyDeclaration": "$honestyDeclaration"}""").as[JsObject]
```

Comment:
This would be better as `Json.obj("honestyDeclaration" -> honestyDeclaration)` because you're creating the JsObject directly rather than parsing a string into a JsValue then casting it to a JsObject.